daemon/src/maker_inc_connections.rs

```diff
@@ -102,43 +111,40 @@ impl Actor {
         let transport_state = Arc::new(Mutex::new(transport_state));
 
         let (read, write) = stream.into_split();
-        let read = FramedRead::new(read, wire::EncryptedJsonCodec::new(transport_state.clone()))
-            .map_ok(move |msg| FromTaker { taker_id, msg })
-            .map(forward_only_ok::Message);
+        let mut read =
+            FramedRead::new(read, wire::EncryptedJsonCodec::new(transport_state.clone()))
+                .map_ok(move |msg| FromTaker { taker_id, msg });
+
+        let this = ctx.address().expect("self to be alive");
+        let taker_msg_channel = self.taker_msg_channel.clone_channel();
+        let read_fut = async move {
+            while let Ok(res) = read.try_next().await {
+                match res {
+                    Some(msg) => taker_msg_channel
+                        .send(msg)
+                        .await
+                        .expect("connection to taker to exist"),
+                    None => continue,
+                }
+            }
 
-        let (out_msg_actor_address, mut out_msg_actor_context) = xtra::Context::new(None);
+            let _ = this.send(ReadFail(taker_id)).await;
+        };
+        taker_tasks.add(read_fut);
 
-        let (forward_to_cfd, forward_to_cfd_fut) =
-            forward_only_ok::Actor::new(self.taker_msg_channel.clone_channel())
-                .create(None)
-                .run();
-        self.tasks.add(forward_to_cfd_fut);
+        let (out_msg, mut out_msg_actor_context) = xtra::Context::new(None);
+        let send_to_socket_actor = send_to_socket::Actor::new(write, transport_state.clone());
 
-        // only allow outgoing messages while we are successfully reading incoming ones
         let heartbeat_interval = self.heartbeat_interval;
```

Comment:
Can we inlined, no longer needs to be separate because the `async move` is gone.